member.php

```diff
@@ -783,7 +783,64 @@
 	{
 		$plugins->run_hooks("member_register_start");
 
-		$validator_extra = '';
+		// JS validator extra
+		if($mybb->settings['maxnamelength'] > 0 && $mybb->settings['minnamelength'] > 0)
+		{
+			$lang->js_validator_username_length = $lang->sprintf($lang->js_validator_username_length, $mybb->settings['minnamelength'], $mybb->settings['maxnamelength']);
+		}
+
+		$validator_javascript = "<script type=\"text/javascript\">
```

Comment:
I've had a think and there's 3 ways of going about this:
1. completely hardcoded (as in the PR currently)
2. main JS in template with conditional snippets hardcoded (as before the PR)
3. completely in templates: main JS in template with conditional snippets in new templates

Before this PR the javascript was fragmented with some hardcoded and the bit that's used all the time in the template. I moved the template code to the PHP file to bring it together. I don't think number 2 is preferable at all so it's a case of 1 vs 3.

Personally I'd say number 1 as I believe this type of javascript shouldn't be in templates because the theme authors shouldn't need to change it as it's not part of the style. Although there is nothing to stop them extending it by adding `<script>` tags.